File: sdks/python/examples/basic_usage.py
"""
Basic usage example for Guardian API Python SDK
"""
from guardian_api import GuardianClient, GuardianAPIError

def main():
    # Initialize client
    client = GuardianClient(base_url="http://localhost:8000")
    
    print("=== Guardian API Python SDK Example ===\n")
    
    # Check health
    print("1. Checking API health...")
    try:
        health = client.health_check()
        print(f"   Status: {health['status']}")
        print(f"   Version: {health['version']}")
        print(f"   Models loaded: {health['models_loaded']}\n")
    except GuardianAPIError as e:
        print(f"   Error: {e}\n")
        return
    
    # Get model info
    print("2. Getting model information...")
    try:
        models = client.get_models()
        for model in models['models']:
            status = "✓" if model['loaded'] else "✗"
            print(f"   {status} {model['name']} (v{model['version']})")
        print()
    except GuardianAPIError as e:
        print(f"   Error: {e}\n")
    
    # Moderate single text
    print("3. Moderating single text...")
    test_texts = [
        "This is a completely normal and harmless message.",
        "Women are terrible at coding and should stick to cooking.",
        "You're an idiot and I hope you fail at everything."
    ]
    
    for text in test_texts:
        try:
            result = client.moderate_text(text)
            score = result['ensemble']['score']
            severity = result['ensemble']['severity']
            primary_issue = result['ensemble']['primary_issue']
            
            print(f"   Text: {text[:50]}...")
            print(f"   Score: {score:.3f} | Severity: {severity} | Issue: {primary_issue}")
            print(f"   Summary: {result['ensemble']['summary']}")
            print()
        except GuardianAPIError as e:
            print(f"   Error: {e}\n")
    
    # Batch moderation
    print("4. Batch moderation...")
    try:
        batch_result = client.moderate_batch(test_texts)
        print(f"   Processed: {batch_result['total_processed']} texts")
        print(f"   Time: {batch_result['processing_time_ms']}ms")
        print()
        
        for i, result in enumerate(batch_result['results']):
            print(f"   Text {i+1}: Score {result['ensemble']['score']:.3f}")
    except GuardianAPIError as e:
        print(f"   Error: {e}\n")
    
    # Context manager usage
    print("5. Using context manager...")
    with GuardianClient(base_url="http://localhost:8000") as client:
        result = client.moderate_text("Test message")
        print(f"   Score: {result['ensemble']['score']:.3f}\n")

if __name__ == "__main__":
    main()

File: sdks/javascript/examples/basic-usage.js
/**
 * Basic usage example for Guardian API JavaScript SDK (CommonJS)
 */
const { GuardianClient, GuardianAPIError } = require('../dist');

async function main() {
  // Initialize client
  const client = new GuardianClient({
    baseUrl: 'http://localhost:8000'
  });

  console.log('=== Guardian API JavaScript SDK Example ===\n');

  // Check health
  console.log('1. Checking API health...');
  try {
    const health = await client.healthCheck();
    console.log(`   Status: ${health.status}`);
    console.log(`   Version: ${health.version}`);
    console.log(`   Models loaded: ${health.models_loaded}\n`);
  } catch (error) {
    if (error instanceof GuardianAPIError) {
      console.log(`   Error: ${error.message}\n`);
    }
    return;
  }

  // Moderate single text
  console.log('2. Moderating text...');
  try {
    const result = await client.moderateText('This is a test message');
    console.log(`   Score: ${result.ensemble.score.toFixed(3)}`);
    console.log(`   Severity: ${result.ensemble.severity}`);
    console.log(`   Primary Issue: ${result.ensemble.primary_issue}\n`);
  } catch (error) {
    if (error instanceof GuardianAPIError) {
      console.log(`   Error: ${error.message}\n`);
    }
  }

  // Batch moderation
  console.log('3. Batch moderation...');
  try {
    const batchResult = await client.moderateBatch([
      'Normal text',
      'Harmful text'
    ]);
    console.log(`   Processed: ${batchResult.total_processed} texts`);
    for (let i = 0; i < batchResult.results.length; i++) {
      console.log(`   Text ${i + 1}: Score ${batchResult.results[i].ensemble.score.toFixed(3)}`);
    }
  } catch (error) {
    if (error instanceof GuardianAPIError) {
      console.log(`   Error: ${error.message}\n`);
    }
  }
}

main().catch(console.error);

